_projects/2024/100419840/100419840.Rmd

```diff
@@ -0,0 +1,598 @@
+
+```{r}
+
+data_filtered <- data_filtered %>%
+  mutate(Fecha = as.Date(paste(Año, Trimestre * 3, "01", sep = "-"), format = "%Y-%m-%d"))
+```
+
+Following the previous steps, it is needed to calculate the maximum value of the appraised value during the housing crisis. Therefore, the data is filtered before 2009 (according to the period highlighted in the original graph). The, it is grouped by region to get the maximum value for each one of them. After that, if the value is missing it is replaced by 0 and since this calculus has been made in a different data frame, the results are finally joined by region.
+
+```{r}
+max_data <- data_filtered %>%
+  filter(Año < 2009) %>%
+  group_by(CCAA) %>%
+  summarise(
+    Valor_Max = max(Valor_Tasado, na.rm = TRUE),
+    .groups = "drop"
+  ) %>%
+  complete(CCAA, fill = list(Valor_Max = 0))  
+
+data_filtered <- data_filtered %>%
+  left_join(max_data, by = "CCAA")
+```
+
+Finally, a variable is aggregated to be able to represent the evolution of housing prices according to the maximum value calculated before. This variable contains the porcentual change in housing prices for the maximum value in the crisis period. Moreover, it has been contemplated that if the maximum value is missing (equal to 0), the Evolution (Evolución) value is also 0.
+
+```{r}
+data_filtered <- data_filtered %>%
+  mutate(
+    Evolucion = if_else(
+      Valor_Max != 0,  
+      ((Valor_Tasado - Valor_Max) / Valor_Max) * 100, 0 ))
+```
+
+
+## Replication
+
+Once the data is prepared, it is time to replicate the graph. The first step is to create a basic graph that includes the data to be represented and the axis variables. In this case, the x-axis represents the date ("Fecha") and the y-axis the Evolution. The data is grouped by CCAA and the color to represent each line is determined by the CCAA they belong to.
+
+```{r}
+mi_grafico_mal <- ggplot(data_filtered, 
+                         aes(x = Fecha, y = Evolucion, group = CCAA)) 
+mi_grafico_mal 
+```
+
+To represent the lines, the command *geom_step* has been used since it represents the data according to the original graph (similar to a stairs/steps). Since there are two groups of CCAA, the first step has been to represent those that are not highlighted/labeled, which are represented in grey.
+
+```{r}
+mi_grafico_mal <- mi_grafico_mal + 
+    geom_step(data = subset(data_filtered, 
+                            !CCAA %in% comunidades_destacadas), 
+                            color = "grey80")
+mi_grafico_mal
+```
+The second part is to represent the highlighted regions, which are those contained in the vector of the previous steps and they are colored by region.
+
+```{r}
+mi_grafico_mal <- mi_grafico_mal + geom_step(data = subset(data_filtered, CCAA %in% comunidades_destacadas), aes(color = CCAA),
```

Comment:
Note that this line is too wide, and goes off the page in the resulting HTML depending on the screen size. It's a good practice to wrap your code so that every line has a maximum length of 80 characters approx. Review the rest of the post to solve these kind of problems, please.